app/src/main/java/com/team41/boromi/controllers/BookController.java

```diff
@@ -0,0 +1,265 @@
+                    Log.d(TAG, "Call backing All Filtered Books");
+                    onSuccessFilterBook(availableBooks, requestedBooks, acceptedBooks, borrowedBooks);
+                } else {
+                    Log.d(TAG, "Book empty");
+                    onFailure(new IllegalArgumentException());
+                }
+            });
+        } else {
+            Log.d(TAG, " Error in one of the columns");
+            onFailure(new IllegalArgumentException());
+        }
+    }
+
+    /**
+     * static method to check null/empty
+     * @param str
+     * @return
+     */
+    public static boolean isNotNullOrEmpty(String str) {
+        if (!str.trim().isEmpty() && str != null) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Adds Book to DB asynchronously. On Success or Failure, it will have a callback to let the ui know
+     * @param owner
+     * @param author
+     * @param ISBN
+     * @param title
+     */
+    public Boolean addBook(String owner, String author, String ISBN, String title) {
+        if(isNotNullOrEmpty(author) && isNotNullOrEmpty(ISBN) && isNotNullOrEmpty(title)) {
+            Book addingBook = new Book(owner, title, author, ISBN);
+            addingBook.setStatus(status.AVAILABLE);
+            addingBook.setWorkflow(workflow.AVAILABLE);
+            executor.execute(() -> {
+                Book result = bookDB.pushBook(addingBook);
+                if (result != null) {
+//                    Log.d(TAG, " book add success");
+                    onSuccessAddBook("Book Add Success");
+                } else {
+                    Log.d(TAG, " book add error");
+                    onFailure(new IllegalArgumentException());
+                }
+            });
+            return true;
+        } else {
+            Log.d(TAG, " Error in one of the columns");
+            onFailure(new IllegalArgumentException());
+            return false;
+        }
+    }
+
+    /**
+     * Edits book description by getting book from BookDB
+     * @param bookID
+     * @param author
+     * @param ISBN
+     * @param title
+     * @return
+     */
+    public Boolean editBook(String bookID, String author, String ISBN, String title){
+        if (isNotNullOrEmpty(author) && isNotNullOrEmpty(ISBN) && isNotNullOrEmpty(title)) {
+            executor.execute(() -> {
+                Book editingBook = bookDB.getBook(bookID);
+                if (editingBook != null) {
+                    editingBook.setTitle(title);
+                    editingBook.setAuthor(author);
+                    editingBook.setISBN(ISBN);
+                    bookDB.pushBook(editingBook);
+                    Log.d(TAG, " book add success");
+                    onSuccessEditBook("Successful edit");
+                } else {
+                    Log.d(TAG, " book add error");
+                    onFailure(new IllegalArgumentException());
+                }
+            });
+            return true;
```

Comment:
Do you want to return true even if editingBook is null?

Reply:
All the methods are void. Not returning boolean now.
a6deb73
9400dd8